File: ItalianRestaurantBackend/src/main/java/com/example/italianrestaurant/delivery/Delivery.java
package com.example.italianrestaurant.delivery;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.Hibernate;

import java.time.LocalDateTime;
import java.util.Objects;

@Entity
@Table(name = "deliveries")
@Getter
@Setter
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Delivery {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String address;
    @Column(nullable = false)
    private String city;
    @Column(nullable = false)
    private String postalCode;
    @Column(nullable = false)
    private String floor;
    private String info;

    @Enumerated(EnumType.STRING)
    private DeliveryOptions deliveryOptions;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Delivery delivery = (Delivery) o;
        return getId() != null && Objects.equals(getId(), delivery.getId());
    }
}

File: ItalianRestaurantFrontend/src/app/models/delivery.ts
export class Delivery {
  address: string
  city: string
  postalCode: string
  floor: string
  info: string
  deliveryOptions: string


  constructor(address: string, city: string, postalCode: string, floor: string, info: string, deliveryOptions: string) {
    this.address = address;
    this.city = city;
    this.postalCode = postalCode;
    this.floor = floor;
    this.info = info;
    this.deliveryOptions = deliveryOptions;
  }

}
